src/services/attachmentService.ts

```diff
@@ -0,0 +1,414 @@
+/* eslint-disable no-param-reassign, no-await-in-loop */
+import cloneDeep from 'lodash/cloneDeep';
+import find from 'lodash/find';
+import flatMap from 'lodash/flatMap';
+import includes from 'lodash/includes';
+import isEmpty from 'lodash/isEmpty';
+import isNull from 'lodash/isNull';
+import map from 'lodash/map';
+import merge from 'lodash/merge';
+import mergeWith from 'lodash/mergeWith';
+import omit from 'lodash/omit';
+import reject from 'lodash/reject';
+import some from 'lodash/some';
+import {
+  FULL,
+  getAttachmentIdToDownload,
+  getContentHash,
+  getEncryptedFilesByAttachmentIndex,
+  getFileContentsAsBuffer,
+  getIdentifierValue,
+  hasAttachments,
+  separateOldAndNewAttachments,
+  shrinkImageHeightIfNeeded,
+  verifyAttachmentPayload,
+} from '../lib/attachmentUtils';
+import BlobFile from '../lib/BlobFile';
+import InvalidAttachmentPayloadError from '../lib/errors/InvalidAttachmentPayloadError';
+import ValidationError from '../lib/errors/ValidationError';
+import fhirValidator from '../lib/fhirValidator';
+import {
+  isAllowedByteSequence,
+  isAllowedFileType,
+  isResizableImageByteSequence,
+  isWithinSizeLimit,
+} from '../lib/fileValidator';
+import Attachment from '../lib/models/fhir/Attachment';
+import { DOCUMENT_REFERENCE } from '../lib/models/fhir/DocumentReference';
+import documentRoutes from '../routes/documentRoutes';
+import createCryptoService from './createCryptoService';
+import recordService from './recordService';
+import { DecryptedFhirRecord, FetchResponse, Params, Record } from './types';
+
+type IAttachment = {
+  isImage?: boolean;
+  hasPreview?: boolean;
+  hasThumb?: boolean;
+  id?: string;
+};
+
+// TODO: Add return Type
```

Comment:
Is this covered in the `any[]`? Or could this be more precise?

Reply:
Done in e5524bb